### PR TITLE
Try to fix the token rotation failure and waitfor cluster status bug

### PR DESCRIPTION
**Pull request by naiming-zededa**
- the token rotation in current code, only does once then wait forever, if in the case of applying this failed, we didn't have chance to retry. In this fix we'll retry every minute
- have a bug in waiting for bootstrap server status, we can fall into the 'else' condition, which may not be server already in 'cluster' status. we can get a wrong cert from server and can not recover
- printout log once a while if we can not https to bootstrap server while waiting
- if not all-pods-ready, not printing the misleading 'applying node labels' message, instead log the 'Not all pods are ready'

### CODE REVIEW

**Comment by andrewd-zededa** [comment]
I like the throttled logging every 30x10sec

File: pkg/kube/cluster-init.sh [hunk 1]
```diff
@@ -641,16 +641,33 @@ get_enc_status() {
     fi
 }
 
+rotate_cluster_token() {
+        local token="$1"
+        /usr/bin/k3s token rotate --new-token "$token"
```
Comment (andrewd-zededa) [inline pkg/kube/cluster-init.sh:646]: Are there any documented error return codes from this path we should handle?

**Review by andrewd-zededa — COMMENTED** [review]
I just have a question about the return codes of "k3s token rotate", otherwise LGTM.

File: pkg/kube/cluster-init.sh (hunk 1 above)
Comment (naiming-zededa) [reply pkg/kube/cluster-init.sh:646]: I have added some logging, to see the output of k3s command using '$?', just as for debugging purpose.

**Comment by naiming-zededa** [comment]
I also modified the 'applying node labels' message in the case of not all-pods-ready condition.
```
2024-12-31 07:45:07 : Not all pods are ready, Continue to wait while applying node labels
2024-12-31 07:45:18 : Not all pods are ready, Continue to wait while applying node labels
```

**Comment by zedi-pramodh** [comment]
I am merging this, to give build to QA

**Review by zedi-pramodh — APPROVED** [review]
LGTM.

**Pull request merged**